packages/runtime/container-runtime/src/summary/summaryGenerator.ts

```diff
@@ -417,7 +422,15 @@ export class SummaryGenerator {
 				return fail("disconnect");
 			}
 			if (waitAckNackResult.result !== "done") {
-				return fail("summaryAckWaitTimeout");
+				// If summary ack was not received, it could just be a transient error. So,
+				// fail with a retriable error which may result in another attempt.
```

Comment:
summaryAck might not have been received because of some transient error, so here we fail with a retriable error to re-attempt another summary.

Reply:
Re-worded it. The re-attempt may not happen depending on how many attempts have happened yet. So, I kept that part of the comment.